daemon/api_debug.go

```diff
@@ -83,6 +84,58 @@ func checkConnectivity(st *state.State) Response {
 	return SyncResponse(status, nil)
 }
 
+type changeTimings struct {
+	DoingTime      time.Duration         `json:"doing-time,omitempty"`
+	UndoingTime    time.Duration         `json:"undoing-time,omitempty"`
+	DoingTimings   []*timings.TimingJSON `json:"doing-timings,omitempty"`
+	UndoingTimings []*timings.TimingJSON `json:"undoing-timings,omitempty"`
+}
+
+func getChangeTimings(st *state.State, changeID string) Response {
+	chg := st.Change(changeID)
+	if chg == nil {
+		return BadRequest("cannot find change: %v", changeID)
+	}
+
+	// lookup for filtering tasks from "timings"
+	changeTasks := make(map[string]bool)
+	for _, t := range chg.Tasks() {
+		changeTasks[t.ID()] = true
+	}
+
+	doingTimingsByTask := make(map[string][]*timings.TimingJSON)
+	undoingTimingsByTask := make(map[string][]*timings.TimingJSON)
+
+	// collect "timings" for task of given change
+	var stateTimings []timings.RootTimingsJSON
+	st.Get("timings", &stateTimings)
+	for _, tm := range stateTimings {
```

Comment:
it seems we need a helper in timings that takes state,  and a filter functions taking tags

Reply:
Implemented now.